### PR TITLE
Enhance OpenWrt / musl libc support

### PR DESCRIPTION
**Pull request by michaelortmann**
Found by: michaelortmann
Patch by: michaelortmann
Fixes: 

One-line summary:
Fix aclocal.m4 under OpenWrt

Additional description (if needed):
1. Don't add -ldl if dlopen is in system lib
2. Enhance Eggdrop Compile FAQ

**Please run misc/runautotools**

Test cases demonstrating functionality (if applicable):

before:
```
$ ./configure [...]
[...]
checking for dlopen in -ldl... no
[...]
checking for Tcl library flags...  -L/home/michael/opt/tcl-8.6.8/lib -ltcl8.6 -ldl  -lpthread
[...]
$ make
[...]
/usr/bin/ld: cannot find -ldl
collect2: error: ld returned 1 exit status
[...]
```

after:
```
$ ./configure
[...]
checking for library containing dlopen... none required
[...]
checking for Tcl library flags... -L/home/michael/opt/tcl-8.6.8/lib -ltcl8.6  -lpthread
[...]
$ make
[...]
All modules compiled.
[...]
```

Hello OpenWrt world:
```
$ grep  PRETTY_NAME /etc/os-release 
PRETTY_NAME="OpenWrt 18.06.1"
$ uname -a
Linux OpenWrt 4.14.63 #0 SMP Wed Aug 15 20:42:39 2018 i686 GNU/Linux
$ /usr/lib/gcc/i486-openwrt-linux-musl/5.4.0/libc.so --version
musl libc (i386)
Version 1.1.19
Dynamic Program Loader
Usage: /usr/lib/gcc/i486-openwrt-linux-musl/5.4.0/libc.so [options] [--] pathname [args]
$ LD_LIBRARY_PATH=/home/michael/opt/tcl-8.6.8/lib ./eggdrop -v
Eggdrop v1.8.3+dragonfly (C) 1997 Robey Pointer (C) 2010-2018 Eggheads
Configure flags: '--with-tcllib=/home/michael/opt/tcl-8.6.8/lib/libtcl8.6.so' '--with-tclinc=/home/michael/opt/tcl-8.6.8/include/tcl.h'
Compiled with: IPv6, handlen=32
```

### CODE REVIEW

**Review by thommey — APPROVED** [review]
LGTM

**Comment by vanosg** [comment]
build

**Comment by vanosg** [comment]
I want to add a little to the docs you added.. let's chat in channel on this when you're around

**Comment by michaelortmann** [comment]
There may be another bashism in here, that i would have to fix...

**Comment by michaelortmann** [comment]
@vanosg: requested doc has been added.

**Comment by vanosg** [comment]
Couldn't have said it better myself. Great work!

File: doc/COMPILE-GUIDE [hunk 1]
```diff
@@ -45,6 +45,7 @@ Last revised: September 20, 2018
          macro assignment' error when I type 'make config' (OpenIndiana)
      12. I get a 'gmake: readlink: Command not found' error when I type 'gmake
          install' (OpenIndiana)
+     13. I get '/usr/bin/ld: cannot find -lpthread' (OpenWrt)
```
Comment (Cizzle) [inline doc/COMPILE-GUIDE]: According to your test case the error is about `/usr/bin/ld: cannot find -ldl`?

**Pull request merged**